ds3-autogen-net/src/main/java/com/spectralogic/ds3autogen/net/generators/typeparsers/BaseTypeParserGenerator.java

```diff
@@ -34,12 +35,31 @@
     public BaseTypeParser generate(
             final ImmutableMap<String, Ds3Type> typeMap) {
         final ImmutableList<TypeParser> typeParsers = toTypeParserList(typeMap);
+        final ImmutableList<String> enumParsers = toEnumList(typeMap);
 
-        return new BaseTypeParser(typeParsers);
+        return new BaseTypeParser(
+                typeParsers,
+                enumParsers);
     }
 
     /**
-     * Converts a list of Ds3Types into a list of TypeParsers
+     * Retrieves the list of enum type names from within a Ds3Types list
+     */
+    protected static ImmutableList<String> toEnumList(final ImmutableMap<String, Ds3Type> typeMap) {
+        if (isEmpty(typeMap)) {
+            return ImmutableList.of();
+        }
+        final ImmutableList.Builder<String> builder = ImmutableList.builder();
```

Comment:
A part of me wants to replace this with a stream implementation that does the same thing.  Functionally it is the same so I'm not sure what it would buy us besides removing some lines of code.